pkg/apis/ingress.adevinta.com/v1beta1/serialization_test.go

```diff
@@ -0,0 +1,65 @@
+package v1beta1_test
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubectl/pkg/scheme"
+
+	"github.com/adevinta/k8s-traffic-controller/pkg/apis/ingress.adevinta.com/v1beta1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFullyFlegedDeserialization(t *testing.T) {
+	serialized := `kind: ClusterIngressServiceDNSWeight
```

Comment:
This could be nicer if kept in a separate file and loaded, so formatting for yaml can be kept in the YAML file. That would ease future changes to the contents of this test file.

Reply:
It is intentional to keep it in the same file, to allow having all the context at a glance.
Here, we are talking about a serialisation test, I valued visualising both the code and the serialized changes